src/main/java/ttt/PerfectPlayer.java

```diff
@@ -0,0 +1,81 @@
+package ttt;
+
+import java.util.*;
+
+public class PerfectPlayer implements Player {
+
+    private String mark;
+    private String opponent;
+    private HashMap<Integer, Integer> scores = new HashMap<>();
+    private Map.Entry<Integer, Integer> bestMove;
+
+    public PerfectPlayer(String mark) {
+        this.mark = mark;
+        this.opponent = mark.equals("X") ? "O" : "X";
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/mollie-c-stephenson/TicTacToe/file/2575923627/issues/source?bid=3284885&fileBranchId=3299578#l14)